src/main/java/com/foundation/search/model/Search.java

```diff
@@ -11,9 +11,249 @@
  */
 
 package com.foundation.search.model;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.common.io.Files;
+import org.apache.commons.io.FileUtils;
 
 public class Search {
-    public Search() {
 
+    /**
+     * searchFiles method performs a search of files and directories by
+     * following next criteria:
+     * Search by Path.
+     * Search by file name.
+     * Search by extension.
+     * Search by hidden property.
+     * Search by read only property.
+     * Search only Directories.
+     * Search by Size
+     * @param criteria Object that belongs to Criteria class, where
+     *                 all the search criteria is defined.
+     * @return results List of Objects that contains the search results.
+     */
+    public List<SearchResult> searchFiles(Criteria criteria){
+
```

Comment:
For consistence, please remove this blank line, since in the other methods there is not a blank line between the method definition and the parameters definition